homeassistant/core.py

```diff
@@ -1749,7 +1749,9 @@ class Service:
 
     def __init__(
         self,
-        func: Callable[[ServiceCall], Coroutine[Any, Any, ServiceResponse] | None],
+        func: Callable[
+            [ServiceCall], Coroutine[Any, Any, ServiceResponse] | ServiceResponse | None
```

Comment:
`ServiceResponse` is already optional.

Although I think we should probably remove that from the type variable and be explicit optional in every place instead.

Reply:
We've discussed the `ServiceResponse` varialbe in the core team and we want to remove `None` from it.

We don't need to do that in this PR though, so I'll approve this as is.